### DISCUSSION TITLE
appium:shouldTerminateApp, does this capability works for simulators as well?

### DISCUSSION THREAD

**Discussion by svermagit** [appium/appium#15737] [Q&A] [answered]
appium:shouldTerminateApp, does this capability works for simulators as well?

**Comment by jlipps** [comment]
Where did you learn about this capability? I'm not aware of it.

**Reply by svermagit** [reply]
This is new capability added in appium 1.21.0 release for iOS

Please see this: https://github.com/appium/appium-xcuitest-driver/pull/1290 

Also this is present in appium documentation:

https://github.com/appium/appium-xcuitest-driver#desired-capabilities

**Reply by jlipps** [reply]
Seems like it works for simulators as well.

**Reply by svermagit** [reply]
Thanks for confirming.

is there any specific test available for this capability, I am not getting the way to use this?

**Reply by jlipps** [reply]
What have you tried already?

**Reply by svermagit** [reply]
Generally what happens in my case every time a simulator gets launch it opens a new session, no previous history is being maintained here. So in that case it is some how difficult to apply assertion with this capability if session has been ended and when again open the simulator. Because by itself the simulator is completely killed on session closed. so tried below code with combination of noReset=true and shouldTerminateApp=true

driver.launchSimulator()
getDriver().closeApp();
		Log.info("App Closed");
		getDriver().launchApp();
		Log.info("App Launched");

**Reply by jlipps** [reply]
why are you using launchSimulator, closeApp, and launchApp? rather than just starting a session with a simulator running? if you don't have appium start a sim for you, it will not close one. and i would stay away from closeApp and launchApp if you can. just put your app details in the capabilities and appium will launch it for you on session start.

**Comment by KazuCocoa** [comment, marked as answer]
The end of the capability calls 
https://developer.apple.com/documentation/xctest/xcuiapplication/1500637-terminate in the session deletion.
How the API works depends on XCTest internal.

You can check if the API calls in appium log and the device log.

**Reply by svermagit** [reply]
Thanks for this useful info. 
We do have some platform specific mechanism to terminate the app on session-end with simulators. In that case no impact can be seen with this capability.
But I will take a look if this capability can be utilized in anyway.
